astropy/modeling/fitting.py

```diff
@@ -1181,81 +1280,81 @@ def __call__(self, model, x, y, z=None, weights=None,
-            sum_sqrs = np.sum(self.objective_function(fitparams, *farg)**2)
-            dof = len(y) - len(init_values)
-            self.fit_info['param_cov'] = cov_x * sum_sqrs / dof
-        else:
-            self.fit_info['param_cov'] = None
+        return init_values, fitparams, cov_x
 
-        if self._calc_uncertainties is True:
-            if self.fit_info['param_cov'] is not None:
-                self._add_fitting_uncertainties(model_copy,
-                                               self.fit_info['param_cov'])
-
-        model_copy.sync_constraints = True
-        return model_copy
 
-    @staticmethod
-    def _wrap_deriv(params, model, weights, x, y, z=None):
-        """
-        Wraps the method calculating the Jacobian of the function to account
-        for model constraints.
-        `scipy.optimize.leastsq` expects the function derivative to have the
-        above signature (parlist, (argtuple)). In order to accommodate model
-        constraints, instead of using p directly, we set the parameter list in
-        this function.
-        """
+class TRFLSQFitter(_NonLinearLSQFitter):
+    """
+    Trust Region Reflective (dogbox or Levenberg-Marqueardt) algorithm
+    and least squares statistic.
 
-        if weights is None:
-            weights = 1.0
+    Attributes
+    ----------
+    method : str
+        The method used to perform the fit (see notes)
+    fit_info :
+        A `scipy.optimize.OptimizeResult` class which contains all of
+        the most recent fit information
 
-        if any(model.fixed.values()) or any(model.tied.values()):
-            # update the parameters with the current values from the fitter
-            fitter_to_model_params(model, params)
-            if z is None:
-                full = np.array(model.fit_deriv(x, *model.parameters))
-                if not model.col_fit_deriv:
-                    full_deriv = np.ravel(weights) * full.T
-                else:
-                    full_deriv = np.ravel(weights) * full
-            else:
-                full = np.array([np.ravel(_) for _ in model.fit_deriv(x, y, *model.parameters)])
-                if not model.col_fit_deriv:
-                    full_deriv = np.ravel(weights) * full.T
-                else:
-                    full_deriv = np.ravel(weights) * full
+    Notes
+    -----
+    The ``method`` determines which fitter algorithm to use:
+        ‘trf’ : Trust Region Reflective algorithm, particularly suitable
+            for large sparse problems with bounds. Generally robust method.
+        ‘dogbox’ : dogleg algorithm with rectangular trust regions, typical
+            use case is small problems with bounds. Not recommended for
+            problems with rank-deficient Jacobian.
+        ‘lm’ : Levenberg-Marquardt algorithm as implemented in MINPACK.
+            Doesn’t handle bounds and sparse Jacobians. Usually the most
+            efficient method for small unconstrained problems.
```

Comment:
This class also needs a `Parameters` section. The notes can probably go in it.

Reply:
See cbe5e83